signer/aws/aws.go

```diff
@@ -22,54 +24,85 @@ import (
 	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
 )
 
-// OpenSearchService AWS OpenSearchService Name
+// OpenSearchService Amazon OpenSearch Service Name
 const OpenSearchService = "es"
 
-// Signer is a interface that will implement opensearchtransport.Signer
+// OpenSearchServerless Amazon OpenSearch Serverless Name
+const OpenSearchServerless = "aoss"
+
+const emptyBodySHA256 = "2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824"
```

Comment:
You can skip the empty body hash since the signer assigns the empty body hash if the request body is null, looking at https://github.com/aws/aws-sdk-go/blob/main/aws/signer/v4/v4.go#L708

Reply:
As far as I can tell that's only if `includeSHA256Header ` is true which isn't for `aoss`, only some s3 services.